### PR TITLE
(LTH-6) Add the source line number to logging messages

### PR DESCRIPTION
**Pull request by parisiale**
Adding the source line number by using the __LINE__ macro.
Updating logging unit tests.

### CODE REVIEW

**Comment by parisiale** [comment]
Tested on cthun-agent, on OS X.
Ex:

2015-04-16 23:36:00.187654 TRACE puppetlabs.cthun_agent.connection:268 - WebSocket TLS initialization event

**Comment by parisiale** [comment]
Created LTH-6.

**Comment by MikaelSmith** [comment]
The commit message should be updated to include the ticket.

The number is low-profile, so I'm not particularly against it, but in Facter we don't really have per-file signatures anymore, so I'd be tempted to have a way to disable it via another macro.

File: logging/src/logging.cc [hunk 1]
```diff
@@ -53,7 +53,7 @@ namespace leatherman { namespace logging {
                 << expr::format_date_time<boost::posix_time::ptime>("TimeStamp", "%Y-%m-%d %H:%M:%S.%f")
                 << " " << left << setfill(' ') << setw(5) << leatherman::logging::log_level_attr
                 << " " << leatherman::logging::namespace_attr
-                << " - " << expr::smessage);
+                << ":" << expr::smessage);
```
Comment (MikaelSmith) [inline logging/src/logging.cc]: If we add a macro to disable printing the line number, I'd move the `:` formatting to line 158.

**Comment by parisiale** [comment]
I've updated the title tag.
I'll look into adding a macro.

**Comment by branan** [comment]
This largely looks pretty good to me. It will have some issues once #13 is merge - the big one being that I think the change to `logging.cpp` will need to be a runtime toggle, not an ifdef.

I'm happy to make those tweaks myself as part of LTH-1, though, and merge this as-is.

**Comment by parisiale** [comment]
Thanks @branan 
If it will get too complicated with LTH-1, I suggest to simply drop this.

**Comment by MikaelSmith** [comment]
Runtime toggle is too bad, but I bet branch-prediction would make it negligible.

**Comment by branan** [comment]
@MikaelSmith yeah. Branch prediction, and/or a good LTO pass.

Also we're doing I/O here, so I don't care that much about an extra branch - should be negligable :)

**Comment by MikaelSmith** [comment]
If we're going to have to make it runtime check anyway, you can drop the last two commits and we'll just merge the first.

**Comment by parisiale** [comment]
I reverted the commits, to leave them in the history.
Let me know if you prefer a force push instead.

**Comment by MikaelSmith** [comment]
Oh, can you force push the commits? I'd rather not keep the intermediate work, and need to fix the 1st commit to reference LTH-6.

**Comment by parisiale** [comment]
Will do.

**Comment by MikaelSmith** [comment]
Travis is taking so long...

I started looking ahead to what we'd do to make logging line numbers configurable per-project that will work with the LTH-1 work - @branan - and ended up with https://github.com/puppetlabs/leatherman/compare/master...MikaelSmith:maint/master/per-project-logging-line-numbers.

**Comment by branan** [comment]
@MikaelSmith my idea was pretty much the same, but instead of overloading functions would pass a boolean and trust the predictor and/or LTO to sort it out for us.

Given that logging code isn't critical-path, I'm not sure we care about whether it gets optimized either way

**Comment by MikaelSmith** [comment]
Ended up with a PR at https://github.com/puppetlabs/leatherman/pull/15.

**Pull request merged**